pkgs/development/python-modules/camply/default.nix

```
@@ -0,0 +1,62 @@
+{ lib
+, stdenv
+, buildPythonPackage
+, fetchPypi
+, click
+, fake-useragent
+, hatchling
+, pandas
+, pydantic
+, python-dotenv
+, pytz
+, pyyaml
+, ratelimit
+, requests
+, rich
+, rich-click
+, tenacity
+}:
+
+buildPythonPackage rec {
+  pname = "camply";
+  version = "0.22.1";
+  format = "pyproject";
+
+  src = fetchPypi {
+    inherit pname version;
+    hash = "sha256-9FW4qdsrpXw9JbZCVB1CyOgvFViFlEBn8sEXiKKuzw4=";
+  };
+
+  postPatch = ''
+    substituteInPlace pyproject.toml \
+      --replace 'requests~=2.28.2' 'requests~=2.31.0'
+  '';
+
+  nativeBuildInputs = [
+    hatchling
+  ];
+
+  propagatedBuildInputs = [
+    click
+    fake-useragent
+    pandas
+    pydantic
+    python-dotenv
+    pytz
+    pyyaml
+    ratelimit
+    requests
+    rich
+    rich-click
+    tenacity
+  ];
+
+  pythonImportsCheckHook = [ "camply" ];
+
```

```suggestion
  nativeCheckInputs = [
    pytestCheckHook
  ];
```

Set up tests explicitly. The project uses pytest.